### ISSUE TITLE
Possibilities to adjust for weather protection?

### ISSUE THREAD

**Issue by kentaochstoffe** [pzinemanas/AudioMoth-Firmware-SPL#1]
Hi! First of all, I wanted to thank you for sharing this code to calculate SPL in audiomoths, great initiative with many applications. We are currently starting up a research project to monitor bird wildlife in Sweden using audiomoths, and we want to test the SPL firmware in this project. I wanted to ask a follow up question concerning weather protection. As I understand it, the firmware calculates SPL based on a "clean" recording, i.e. without any wind/weather protection. For someone not familiar with coding, is it possible to somehow tweak the code to compensate for such protections? (see e.g. https://www.openacousticdevices.info/support/device-support/sound-transmission-with-and-without-cases-comparison-with-sm2bat )

Thanks again for sharing. Best regards

**Comment by pzinemanas** [comment]
Hi! It is good to know that this repository is useful for your research project. This is strongly related to the [microphone response compensation](https://github.com/pzinemanas/AudioMoth-Firmware-SPL/blob/master/notebooks/Mic_compensation_filter.ipynb). For now, this compensation is performed using an IIR filter and therefore uses few operations. This allows an implementation of this filter in real time, but clearly the result is not the best. We should improve this compensation by getting the 'true' frequency response (e.g. see Section 7.1 from "[The Implementation of Low-cost Urban Acoustic Monitoring Devices](https://arxiv.org/pdf/1605.08450.pdf)" ). To do this, we should use a FIR filter. This involves much more operations and almost certainly could not be implemented in real time. The same applies if you use a windshield or a case.

In short, first I would find the frequency response of the microphone+protection. Then I'd consider two options:
1) Compensate the signals when the Audiomoth is not recording. This means that you load the audio files from the SD, compensate the signal and calculate the SPL while the Audiomoth is not recording.
2) Compensate the signal and calculate the SPL values offline. This means doing all signal processing and SPL calculation in your computer/server once you get the audio files from the Audiomoth. In this case, you don't need this firmware ;). Instead you have to implement the same operations in a Python script, for instance.

I hope this answers your question, regards!

**Comment by kentaochstoffe** [comment]
Hi! Thank you very much, We will consider these options, possibly also trying the firmware as is.
Thanks again, 
/Gunnar
